From 511324a538a034f7f0766b7689ed50ed4e21ee43 Mon Sep 17 00:00:00 2001
From: Daniel Drake <dsd@gentoo.org>
Date: Thu, 17 Sep 2009 23:49:53 +0545
Subject: [PATCH] Document breakage from internal handle structure change

Code exists which expects usb_dev_handle to always have the same
contents as it did in libusb-0.1.12. Such code will not work with the
compat layer since the structure has changed.
---
 README |    5 +++++
 1 files changed, 5 insertions(+), 0 deletions(-)

diff --git a/README b/README
index 15f17a5..97ab68d 100644
--- a/README
+++ b/README
@@ -29,6 +29,11 @@ Known quirks/differences from libusb-0.1:
  5. Error codes returned on certain events may not exactly match the error
     codes returned by libusb-0.1. Patches accepted to bring us closer to the
     behaviour of libusb-0.1 on Linux.
+ 6. The internal structure of usb_dev_handle is different from libusb-0.1. Of
+    course, since this is a purely internal structure, that shouldn't cause
+    any problems. In reality, some libusb-0.1 users make assumptions about what
+    is inside (bad programming practice) and those assumptions are no longer
+    true, resulting in broken software.
 
 libusb homepage:
 http://libusb.sourceforge.net
-- 
1.7.2.2

